src/plots/cartesian/axes.js

```diff
@@ -1203,6 +1214,48 @@ axes.calcTicks = function calcTicks(ax, opts) {
     return ticksOut;
 };
 
+function syncTicks(ax) {
+    var rng = Lib.simpleMap(ax.range, ax.r2l);
+    var exRng = expandRange(rng);
+    var tickMin = Math.min(exRng[0], exRng[1]);
+    var tickMax = Math.max(exRng[0], exRng[1]);
+
+    // get the overlaying axis
+    var baseAxis = ax._mainAxis;
+
+    var ticksOut = [];
+    if(baseAxis && baseAxis._vals.length > 0) {
+        for(var isMinor = 0; isMinor <= 1; isMinor++) {
+            if(isMinor && !ax.minor) continue;
+
+            for(var i = 0; i < baseAxis._vals.length; i++) {
+                // get the position of the every tick
+                var pos = baseAxis.l2p(baseAxis._vals[i].x);
+                // get the tick for the current axis based on position
+                var vali = ax.p2l(pos);
+                if(vali > tickMin && vali < tickMax) {
+                    var obj = axes.tickText(ax, vali);
+
+                    if(isMinor) {
```

Comment:
Minor ticks don't have labels. 
There might be no need to call `tickText` to generate the label and then set it to blank below!
Could we by pass minor ticks earlier in the process?

Reply:
Yep, fixed.

Reply:
That's the same behaviour used to tickmode === array, and I'm not sure what's the best approach for that.

Reply:
Then that's OK. Resolving.